cmd/geth/consolecmd_test.go

```diff
@@ -15,3 +15,139 @@
 // along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
 
 package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// Tests that a node embedded within a console can be started up properly and
+// then terminated by closing the input stream.
+func TestConsoleWelcome(t *testing.T) {
+	coinbase := "0x8605cdbbdb6d264aa742e77020dcbc58fcdce182"
+
+	// Start a geth console, make sure it's cleaned up and terminate the console
+	geth := runGeth(t, "--nat", "none", "--nodiscover", "--etherbase", coinbase, "-shh", "console")
+	defer geth.expectExit()
+	geth.stdin.Close()
+
+	// Make sure that the welcome message contains all the necessary infos
+	blob, _ := ioutil.ReadAll(geth.stdout)
+	output := string(blob)
+
+	if want := "Welcome"; !strings.Contains(output, want) {
```

Comment:
No, but the console shouldn't output ANSI color escapes unless the output is a terminal.